src/convenience/session.ts

```diff
@@ -76,35 +84,52 @@ export interface StorageAdapter<T> {
      * Reads a value for the given key from the storage. May return the value or
      * undefined, or a promise of either.
      */
-    read: (key: string) => MaybePromise<T | undefined>;
+    read: (key: Key) => MaybePromise<T | undefined>;
     /**
      * Writes a value for the given key to the storage.
      */
-    write: (key: string, value: T) => MaybePromise<void>;
+    write: (key: Key, value: T) => MaybePromise<void>;
     /**
      * Deletes a value for the given key from the storage.
      */
-    delete: (key: string) => MaybePromise<void>;
+    delete: (key: Key) => MaybePromise<void>;
     /**
      * Checks whether a key exists in the storage.
      */
-    has?: (key: string) => MaybePromise<boolean>;
+    has?: (key: Key) => MaybePromise<boolean>;
     /**
-     * Lists all keys.
+     * Lists all keys via an iterator. Optionally, only lists keys that start
+     * with a given prefix.
      */
-    readAllKeys?: () => Iterable<string> | AsyncIterable<string>;
+    keys?: (prefix?: Key) => MaybeAsyncIterable<CascadingKey>;
     /**
-     * Lists all values.
+     * Lists all values via an iterator. Optionally, only lists values for keys
+     * that start with a given prefix.
      */
-    readAllValues?: () => Iterable<T> | AsyncIterable<T>;
+    values?: (prefix?: Key) => MaybeAsyncIterable<T>;
     /**
-     * Lists all keys with their values.
+     * Lists all key-value pairs via an iterator. Optionally, only lists pairs
+     * with keys that start with a given prefix.
      */
-    readAllEntries?: () =>
-        | Iterable<[key: string, value: T]>
-        | AsyncIterable<[key: string, value: T]>;
+    entries?: (prefix?: Key) => MaybeAsyncIterable<KV<T>>;
```

Comment:
These methods need to be reintroduced in order to prevent a breaking change. We can deprecate them in favour of the new methods.